prime-router/src/test/kotlin/azure/ApiKeysFunctionsTest.kt

```diff
@@ -373,6 +373,31 @@ class ApiKeysFunctionsTest {
             val response = ApiKeysFunctions().post(httpRequestMessage, organization.name)
             assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
         }
+
+        @Test
+        fun `Test returns the error if one is encountered persisting the key`() {
```

Comment:
Test covers the situation where the putSetting fails and confirms that the client sees the error